beetmoverscript/script.py

```diff
@@ -82,6 +89,27 @@
     await retry_download(context=context, url=source, path=abs_file_path)
     await retry_upload(context=context, destinations=destinations, path=abs_file_path)
 
+    if context.task['payload']['update_manifest']:
+        enrich_balrog_manifest(context, abs_file_path, locale)
```

Comment:
It's conditioned by 'update_manifest' being set to True. Only the tasks with signed artifacts have the flag set to `True`, hence nothing but CMARs or APK will reach this line.